src/matchRoute.js

```diff
@@ -12,12 +14,11 @@ export function matchRoute(resolvedRoutesConfig, pathMatch) {
     resolvedRoutesConfig.base.length - 1
   );
 
-  if (pathMatch.startsWith(baseWithoutSlash)) {
-    result.routes = recurseRoutes(
-      pathMatch,
-      resolvedRoutesConfig.base,
-      resolvedRoutesConfig.routes
-    );
+  if (pathMatch.indexOf(baseWithoutSlash) === 0) {
+    const origin = inBrowser ? window.location.origin : "http://localhost";
```

Comment:
Matching routes now requires creating a URL/Location object, since we're using activeWhen functions instead of path.startsWith. Only the pathname matters, but to generate a URL we have to have a URL origin. So on the server side I'm setting it to http://localhost just as a "we need something to create this URL"